File: t1-read.c

#include "var.h"

int main(int argc, char **argv){
	data_t t1;
	if(argc != 2){
		printf(" Usage is 1) output filename \n");
		exit(0);
	}	
	printf(" size--%ld \n", sizeof t1);
	
	FILE *fp = fopen(argv[1], "rb");
	if(fp != NULL){
		memset(&t1, 0, sizeof t1);
		fread(t1.buffer, sizeof t1, 1, fp);
		getvale(t1.L.x_t, 150000, 1);	
		getvale(t1.L.y_t, 150000, 1);	
		getvale(t1.L.z_t, 60000, 1);	
		fclose(fp);	
	}else{
		printf(" file[%s] not open error no %d \n", argv[1], errno);
		exit(-1);
	}
		
	return 0;
}

File: t1-read.py
import sys
import bitstruct

def getvalue(val, offset, scale):
    temp = (val*scale)-offset
    print "val- %d offset- %d scale- %d result- %f" % (val, offset, scale, temp)
    return 


if __name__ == "__main__":
    if len(sys.argv) != 2:
            print " Usage python script.py 1)filename"
            sys.exit()
    f = open(sys.argv[1], "rb")
    if f:
        data = f.read()
        data = data[::-1]
        unpack_data = bitstruct.unpack("u18u19u19", data)
        getvalue(unpack_data[0], 60000, 1)
        getvalue(unpack_data[1], 150000, 1)
        getvalue(unpack_data[2], 150000, 1)


    else:
        print "file open error %s "%(sys.argv[1])
